routers/api/v1/repo/issue.go

```diff
@@ -414,7 +414,7 @@ func UpdateIssueDeadline(ctx *context.APIContext, form api.EditDeadlineOption) {
 	var deadline time.Time
 	if form.Deadline != nil && !form.Deadline.IsZero() {
 		deadline = time.Date(form.Deadline.Year(), form.Deadline.Month(), form.Deadline.Day(),
-			23, 59, 59, 0, form.Deadline.Location())
+			0, 0, 0, 0, setting.UILocation)
```

Comment:
This reverses another fix for a bug, in reality we shouldn't be using exact times for deadlines but instead we should be using days.

Reply:
In my projects I persist nomally a date as a timestamp with time set to zero as UTC. During load I add the timezone without changing the time (still zero).

In this change I try to do this with what I understand. But this needs more deeper changes on the db-engine. For me go-lang and the used tools are currently "magic".